libvast/src/query_queue.cpp

```diff
@@ -129,55 +129,56 @@ query_queue::activate(const uuid& qid, uint32_t num_partitions) {
   return caf::none;
 }
 
-// NOLINTNEXTLINE
 std::optional<query_queue::entry> query_queue::next() {
   if (partitions.empty())
     return std::nullopt;
-  auto result = std::move(partitions.back());
-  partitions.pop_back();
-  entry active = {result.partition, 0ull, {}};
-  entry inactive = {result.partition, 0ull, {}};
-  std::partition_copy(
-    std::make_move_iterator(result.queries.begin()),
-    std::make_move_iterator(result.queries.end()),
-    std::back_inserter(active.queries), std::back_inserter(inactive.queries),
-    [&](const auto& qid) {
-      auto it = queries_.find(qid);
-      if (it == queries_.end()) {
-        VAST_WARN("index tried to access non-existant query {}", qid);
-        // Consider it inactive.
-        return false;
+  while (!partitions.empty()) {
```

Comment:
I think the if-clause above can be removed outright, it's already covered by the while-loop.